### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by MrDOS**
At risk of burying the lede, the important change here is bumping `golang.org/x/sys`, because this adds native support for Darwin on ARM64. With the old version, building on an Apple Silicon Mac fails:

```
$ go build ./cmd/...
# golang.org/x/sys/unix
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/syscall_darwin.1_13.go:25:3: //go:linkname must refer to declared function or variable
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/zsyscall_darwin_arm64.1_13.go:27:3: //go:linkname must refer to declared function or variable
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/zsyscall_darwin_arm64.1_13.go:40:3: //go:linkname must refer to declared function or variable
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/zsyscall_darwin_arm64.go:28:3: //go:linkname must refer to declared function or variable
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/zsyscall_darwin_arm64.go:43:3: //go:linkname must refer to declared function or variable
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/zsyscall_darwin_arm64.go:59:3: //go:linkname must refer to declared function or variable
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/zsyscall_darwin_arm64.go:75:3: //go:linkname must refer to declared function or variable
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/zsyscall_darwin_arm64.go:90:3: //go:linkname must refer to declared function or variable
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/zsyscall_darwin_arm64.go:105:3: //go:linkname must refer to declared function or variable
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/zsyscall_darwin_arm64.go:121:3: //go:linkname must refer to declared function or variable
../../go/pkg/mod/golang.org/x/sys@v0.0.0-20200302150141-5c8b2ff67527/unix/zsyscall_darwin_arm64.go:121:3: too many errors
```

With this new version, the build succeeds, and produces a useful executable.

While I was making this change anyway, I thought it would be useful to attempt to bump other dependencies.
* This meant resolving a minor breaking change in `github.com/ekzhu/minhash-lsh`.
* I also wasn't able to bump `gonum.org/v1/gonum` past v0.8.2 because of a breaking change in v0.9.0. See the commit message of f4552ce959319c9bf0da4475678457fd317a98f2 for details.

### CODE REVIEW

**Comment by bzz** [comment]
Thank you for catching and fixing this @MrDOS 🙇 
Re-opened to trigger the CI.

**Comment by bzz** [comment]
@MrDOS could you please rebase it on the latest master, now that win CI profile was removed?

Also, there seems to be a wired issue with linter profile, do you know what might be the reason? In #23 it seems to be working just fine.

>   level=warning msg="[runner] Can't run linter goanalysis_metalinter: buildir: failed to load package goarch: could not load export data: cannot import \"internal/goarch\" (unknown iexport format version 2), export data is newer version - update tool"

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     - name: Run GolangCI-Lint 
       uses: golangci/golangci-lint-action@v2
       with:
-        version: v1.47.0
+        version: v1.50.0
```
Comment (bzz) [inline .github/workflows/lint.yml:13]:
```suggestion
        version: latest
        only-new-issues: true
```

While we are at it, I think it is worth to mute linter on warnings not relevant to the PR and it should be fine to use it's latest version.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@ jobs:
     - name: Run GolangCI-Lint 
       uses: golangci/golangci-lint-action@v2
```
Comment (bzz) [inline .github/workflows/lint.yml:11]:
```suggestion
      uses: golangci/golangci-lint-action@v3
```

**Review by bzz — CHANGES_REQUESTED** [review]

**Comment by bzz** [comment]
Thank you very much for digging deeper into it, your analysis makes perfect sense and Go 1.17 as a target sounds very reasonable.

**Comment by bzz** [comment]
I wonder, why Race CI profile takes 40min and rolling... 🤔  
Oh, my local machine tests take 2min and indeed [claimed runtime overhear](https://go.dev/doc/articles/race_detector#Runtime_Overheads) of `execution time by 2-20x` gives smth like that..

Makes me wonder if that should be something optional rather than mandatory

**Comment by MrDOS** [comment]
Oh, cool, I'd never noticed that the golangci-lint action could take “latest” as a tool version. That's convenient.

It looks like v3 of the action wants the `actions/setup-go` action to be explicitly invoked to install Go in the runner environment. I'll make that change.

Were there any other changes you wanted me to make?

**Comment by bzz** [comment]
> It looks like v3 of the action wants the `actions/setup-go` action to be explicitly invoked to install Go in the runner environment. I'll make that change.

Oh, I see. But it also seems to be working alright now, doesn't it?

> Were there any other changes you wanted me to make?

No, I've committed the suggestions and just forgot to change the review status 😉

**Review by bzz — APPROVED** [review]

**Comment by MrDOS** [comment]
> But it also seems to be working alright now, doesn't it?

Yeah, I think it's just using whatever version of Go is on the `PATH` in the runner. That's probably fine.

It does seem a little weird that the lint action runs tests, too, but also not actively harmful.

Thanks for the green checkmark! I'm happy with this if you're happy with this.

**Comment by bzz** [comment]
> It does seem a little weird that the lint action runs tests, too

Thank you for bringing this up, indeed, this does not seem to be necessary. On the second thought, not relying on container's go version may be a good idea too.

Thank you very much for the improvements! Will merge and 🔪 a new release tomorrow.

**Pull request merged**